jlm/llvm/ir/operators/Store.cpp

```diff
@@ -33,6 +33,40 @@ StoreNonVolatileOperation::copy() const
   return std::unique_ptr<jlm::rvsdg::operation>(new StoreNonVolatileOperation(*this));
 }
 
+[[nodiscard]] size_t
+StoreNonVolatileOperation::NumMemoryStates() const noexcept
+{
+  return nresults();
+}
+
+[[nodiscard]] const StoreNonVolatileOperation &
+StoreNonVolatileNode::GetOperation() const noexcept
+{
+  return *util::AssertedCast<const StoreNonVolatileOperation>(&operation());
+}
+
+[[nodiscard]] StoreNode::MemoryStateInputRange
+StoreNonVolatileNode::MemoryStateInputs() const noexcept
+{
+  if (NumMemoryStates() == 0)
+  {
+    return { MemoryStateInputIterator(nullptr), MemoryStateInputIterator(nullptr) };
+  }
+
+  return { MemoryStateInputIterator(input(2)), MemoryStateInputIterator(nullptr) };
+}
+
+[[nodiscard]] StoreNode::MemoryStateOutputRange
+StoreNonVolatileNode::MemoryStateOutputs() const noexcept
+{
+  if (NumMemoryStates())
+  {
+    return { MemoryStateOutputIterator(nullptr), MemoryStateOutputIterator(nullptr) };
```

Comment:
This is backwards, no? If there are memory states, the iterator pair should not start with nullptr?

Reply:
Well spotted!